dropwizard-core/src/main/java/io/dropwizard/server/AbstractServerFactory.java

```diff
@@ -489,14 +488,16 @@ protected Handler createAppServlet(Server server,
         handler.addFilter(ThreadNameFilter.class, "/*", EnumSet.of(DispatcherType.REQUEST));
         serverPush.addFilter(handler);
         if (jerseyContainer != null) {
-            String urlPattern = jerseyRootPath;
-            if (!urlPattern.endsWith("*") && !urlPattern.endsWith("/")) {
-                urlPattern += "/";
+            if (jerseyRootPath.isPresent()) {
+                String urlPattern = jerseyRootPath.get();
+                if (!urlPattern.endsWith("*") && !urlPattern.endsWith("/")) {
```

Comment:
What do you think about moving this logic for "normalizing" `urlPattern` to `JerseyEnvironment#setUrlPattern`, if we expect users to invoke this method from the application class?